File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/NetworkErrorNotification.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getAndReplaceText;
import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.services.PluginSettings;
import org.jetbrains.annotations.NotNull;

public class NetworkErrorNotification extends Notification {

  @NotNull
  private final Exception exception;

  /**
   * Constructs a notification that notifies the user of an IO error arising from the HTTP client.
   *
   * @param exception An exception that caused this notification.
   */
  public NetworkErrorNotification(@NotNull Exception exception) {
    super(
        PluginSettings.A_PLUS,
        getText("notification.NetworkErrorNotification.title"),
        getAndReplaceText("notification.NetworkErrorNotification.content",
            exception.getMessage()),
        NotificationType.ERROR);
    this.exception = exception;
  }

  @NotNull
  public Exception getException() {
    return exception;
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/NetworkErrorNotification.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

class NetworkErrorNotification
/**
 * Constructs a notification that notifies the user of an IO error arising from the HTTP client.
 *
 * @param exception An exception that caused this notification.
 */(val exception: Exception) : Notification(
    PluginSettings.A_PLUS,
    MyBundle.message("notification.NetworkErrorNotification.title"),
    MyBundle.message(
        "notification.NetworkErrorNotification.content",
        exception.message!!
    ),
    NotificationType.ERROR
)
